igvm/hypervisor.py

```diff
@@ -828,6 +835,48 @@ def device_to_netcat(self, device, size, listener):
             .format(device, size, listener[0], listener[1])
         )
 
+    @contextmanager
+    def xfsrestore(self, vm: VM, transaction: Transaction = None):
+        device = self.get_volume_by_vm(vm).path()
+        port = self.netcat_port(device)
+        mount_dir = self.format_vm_storage(vm, transaction)
+
+        # Check xfsrestore exists, since problem is hidden with
+        # 'nohup', '&' and 'pty=False'
+        self.run('xfsrestore --help', silent=True)
```

Comment:
We are never checking for any other programs as we expect them to be deployed by Puppet. Why do we need to check for this one? And if we go with the way of checking, we should add even more checks, preferrably in another PR, there was even a ticket somewhere to check if it is possible to SSH to a VM before igvm starts, that seems a good idea to implemment too.

Reply:
I agree with @TuxPowered42 here. 

We have certain configuration requirements to a environment such as that some software is installed. This is part of the configuration management and should not be checked here. 

Adding such checks can be useful when having a transition from one OS to another (Stretch to Buster) in combination with conditionals but there is no need for this here.